hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -414,13 +408,12 @@ public void notifyContainerReportProcessing(final boolean isFullReport,
   }
 
   @Override
-  public void deleteContainer(final ContainerID id)
+  public void deleteContainer(final ContainerID cid)
       throws IOException {
-    final HddsProtos.ContainerID cid = id.getProtobuf();
     lock.lock();
     try {
       if (containerExist(cid)) {
-        containerStateManager.removeContainer(cid);
+        containerStateManager.removeContainer(cid.getProtobuf());
```

Comment:
Same question here on lock vs. unnecessary call.

Reply:
Same here - the `getProtobuf()` call is inside the the lock, but before it was outside it.